content/blog/logging-docker-elk.md

```diff
@@ -0,0 +1,93 @@
+  ```
+
+5. Send some logs to Elasticsearch (see below)
+6. Visit <http://localhost:5601> (assuming you’re not running a proxy - also see below)
+7. Log in using username `elastic` and the password you noted earlier
+8. View logs, create visualisations/alerts/dashboards etc
+
+We make some
+[changes to the default configuration](https://github.com/deviantony/docker-elk/compare/main...reside-ic:main) as
+follows:
+
+- We don’t run an Elasticsearch cluster (yet) so we don’t need to
+  expose [port 9300](https://discuss.elastic.co/t/what-are-ports-9200-and-9300-used-for/238578)
+- We’re not currently using the Elasticsearch keystore, so we’ve removed the bootstrap password
+- We’re using Filebeat and Journalbeat in their default configurations to send logs directly to Elasticsearch, so we
+  don’t need to run Logstash - which reduces operational complexity
+- We ensure that Elasticsearch and Kibana restart automatically if the Docker host is rebooted
+- We expose the Kibana web interface via SSL using
+  a [separate Nginx proxy](https://github.com/reside-ic/logs/blob/main/docker-compose.override.yml) container, so we
+  don’t need to expose port 5601 to the host
+- As we use [Vault](https://www.vaultproject.io/) for password management we pass the Kibana password as an environment
+  variable rather than embedding it in the configuration file
+- We only use the non-commercial features of the stack
+
+Once you’ve got Elasticsearch up and running you can set up one or more [Beats](https://www.elastic.co/beats/) agents to
+forward logs from your containers. These too can
+be [run in containers](https://github.com/reside-ic/beats/blob/main/docker-compose.yml). Our applications use the
+Docker `json-file` and `journald` logging drivers, so we use Filebeat and Journalbeat - our setup is
+documented [here](https://github.com/reside-ic/beats). Again, we use Vault for credential storage, but for a simple
+deployment you could store the secrets in a suitably secured Docker Compose `.env` file. These agents talk directly to
+Docker (via volumes and/or socket) so don’t need to be on the same Docker network as the relevant containers.
+
+# Next steps
+
+This will give you a minimal setup - it’s worth ensuring the following for a robust, scalable and secure system:
+
+- Users/roles with access to Kibana are suitably restricted
+- Your Elasticsearch and Kibana Docker volumes are backed-up
+- Sensitive information is either not logged by your applications or is shipped with suitable security (e.g.
+  [using SSL](https://www.elastic.co/guide/en/beats/filebeat/current/configuration-ssl.html)) and appropriately access
+  controlled. Obviously information such as credentials should never be logged.
+- Log messages are suitably structured, so they can be easily searched/filtered
+- Related containers/services share a common Docker Compose project name, Docker container tag etc, so that they can be
+  grouped as one unit within Elasticsearch/Kibana
+
```

Comment:
Can you add a link to our final repo? https://github.com/reside-ic/logs

Reply:
Done